File: Backend/pki-system/src/main/java/com/ftnteam11_2025/pki/pki_system/certificates/dto/CertificateDetailsDTO.java
package com.ftnteam11_2025.pki.pki_system.certificates.dto;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.UUID;

@Getter
@Setter
@Builder
public class CertificateDetailsDTO {
    private UUID id;
    private SubjectIssuerDTO issuer;
    private SubjectIssuerDTO subject;
    private Date validFrom;
    private Date validTo;
    private String certificateKey;
    private String publicKey;
}

File: Frontend/src/app/certificates/model/CertificateDetailsDTO.ts
import { CertificateStatus } from './CertificateStatus';
import { SubjectIssuerDTO } from './SubjectIssuerDTO';

export interface CertificateDetailsDTO {
  id: string;

  issuer: SubjectIssuerDTO;

  subject: SubjectIssuerDTO;
  validFrom: Date;
  validTo: Date;

  certificateKey: string;
  publicKey: string;
  status: CertificateStatus;
}
